mxfusion/modules/gp_modules/svgp_classification.py

```diff
@@ -0,0 +1,424 @@
+
+    v_f = mx.nd.expand_dims(aux, axis=-2) + F.sum(F.square(tmp), -2)
+
+    return mu_f, v_f
+
+
+def quadrature(F, f, dtype, mu_f, v_f, y=None, gh_points=None):
+    if gh_points is None:
+        gh_points = 20
+    gh_x, gh_w = np.polynomial.hermite.hermgauss(gh_points)
+    gh_w = gh_w / np.sqrt(np.pi)
+    gh_x = mx.nd.array(gh_x, dtype=dtype)
+    gh_w = mx.nd.array(gh_w, dtype=dtype)
+    gh_x = mx.nd.expand_dims(gh_x, axis=0)
+    gh_w = mx.nd.expand_dims(gh_w, axis=0)
+
+    shape = mu_f.shape
+
+    mu_f = mx.nd.reshape(mu_f, (-1, 1))
+    v_f = mx.nd.reshape(v_f, (-1, 1))
+
+    X = mx.nd.broadcast_mul(gh_x, mx.nd.sqrt(2. * v_f)) + mu_f
+
+    if y is not None:
+        y = tuple([mx.nd.reshape(y, (-1, 1))])
+    else:
+        y = tuple([])
+
+    out_f = f(X, *y)
+    out = F.linalg.gemm2(out_f, gh_w, transpose_b=True)
+
+    return out.reshape(shape)
+
+
+# TODO: Use Bernoulli logpdf plus link function
+def dec_log_pdf(p, y):
+    y_sign = 2 * y.flatten() - 1.0
+    p = mx.nd.sigmoid(y_sign * p)
+    p = mx.nd.clip(p, 1e-9, 1. - 1e-9)  # for numerical stability
+    return mx.nd.log(p)
+
+
+# TODO: Use the Bernoulli mean function plus link function
+def dec_mean_lik(x):
+    p = mx.nd.sigmoid(x)
+    p = mx.nd.clip(p, 1e-9, 1. - 1e-9)  # for numerical stability
+    return p
+
+
+# TODO: Use link function
+def get_confidence_alpha(mu_f, v_f, alpha=0.95):
+    if alpha <= 0.5:
+        raise ValueError("alpha should be higher than 0.5. Instead got alpha={}.".format(alpha))
+    lb = mx.nd.sigmoid(norm.ppf(1 - alpha) * mx.nd.sqrt(v_f) + mu_f)
+    ub = mx.nd.sigmoid(norm.ppf(alpha) * mx.nd.sqrt(v_f) + mu_f)
+    return lb, ub
+
+class SVGPClassificationLogPdf(VariationalInference):
+    """
+    The inference algorithm for computing the variational lower bound of the stochastic variational Gaussian process with Bernoulli likelihood.
+    """
+    def __init__(self, model, posterior, observed, lik_func, jitter=0.0):
+        super(SVGPClassificationLogPdf, self).__init__(
+            model=model, posterior=posterior, observed=observed)
+        self.log_pdf_scaling = 1
+        self.jitter = jitter
+        self.lik_func = lik_func
+
+    def compute(self, F, variables):
+        X = variables[self.model.X]
+        Y = variables[self.model.Y]
+        Z = variables[self.model.inducing_inputs]
+
+        mu = variables[self.posterior.qU_mean]
+        S_W = variables[self.posterior.qU_cov_W]
+        S_diag = variables[self.posterior.qU_cov_diag]
+
+        D = Y.shape[-1]
+        M = Z.shape[-2]
+
+        mean_prior_f = None
+        mean_prior_u = None
+        if self.model.mean_func is not None:
+            mean_prior_f = self.model.mean_func(F, X)
+            mean_prior_u = self.model.mean_func(F, Z)
+
+        kern = self.model.kernel
+        kern_params = kern.fetch_parameters(variables)
+
+        Kuu = kern.K(F, Z, **kern_params)
+        if self.jitter > 0.:
```

Comment:
I modified it so  now it checks for <0 in the init. I am keeping the >0 since for jitter==0 there is not need to do the F.eye(M, dtype=Z.dtype) * self.jitter operation.